app/src/main/java/com/mithridat/nonoconverter/backend/NonogramSolver.java

```diff
@@ -0,0 +1,154 @@
+package com.mithridat.nonoconverter.backend;
+
+import java.util.ArrayList;
+
+/**
+ * Nonogram solving class
+ */
+class NonogramSolver {
+
+    /**
+     * Show what numbers are ready
+     */
+    private ArrayList<Boolean> _left, _top;
+
+    /**
+     * Nonogram
+     */
+    private Nonogram _nono;
+
+    /**
+     * Field
+     */
+    private Field _field;
+
+    /**
+     * Constructor by the nonogram
+     *
+     * @param nono - nonogram
+     */
+    NonogramSolver(Nonogram nono) {
+        _nono = nono;
+    }
+
+    /**
+     * Method for setting nonogram
+     *
+     * @param nono - nonogram
+     */
+    public void setNonogram(Nonogram nono) {
+        _nono = nono;
+    }
+
+    /**
+     * Method for solving nonogram
+     *
+     * @return true, if solution was founded
+     *         false, otherwise
+     */
+    public boolean solve() {
+
+    }
+
+    /**
+     * Method for initialising class fields before solution start
+     */
+    private void init() {
+
+    }
+
+    /**
+     * Method for checking solution
+     *
+     * @return true, if solution is correct
+     *         false, otherwise
+     */
+    private boolean check() {
+
+    }
+
+    /**
+     * Implementation of simple boxes algorithm
+     *
+     * @param ind - index of row or column
+     * @param type - row or column
```

Comment:
Is it possible to fixate somewhere, what values ​​have a column type and a row type? `enum` or `static private int`, for example?